template/src/pageComponents/signature-mint/ContractDemo.tsx

```diff
@@ -67,40 +61,43 @@ export default function SignatureMintDemo() {
   /**
    * Free Mint Contract Logic
    */
-  const { config: freeMintConfig } = usePrepareContractWrite({
+  const { data: freeMintConfig } = useSimulateContract({
     // TODO: the chainId should be dynamic
     address: contractAddress,
     abi: contract.abi,
     functionName: 'freeMint',
     args: address ? [address, debouncedSigValue] : undefined,
-    enabled: signature.length > 0,
+    query: {
+      enabled: signature.length > 0,
+    },
   });
-  const { write: freeMint } = useContractWrite(freeMintConfig);
+  const { writeContract: freeMint } = useWriteContract();
 
   /**
    * Paid Mint Contract Write Logic
    */
-  const { config: paidMintConfig } = usePrepareContractWrite({
+  const { data: paidMintConfig } = useSimulateContract({
     // TODO: the chainId should be dynamic
     address: contractAddress,
     abi: contract.abi,
     functionName: 'mint',
     args: [address],
     value: parseEther('0.0001'), // You should read the contract, however, setting this to value to prevent abuse.
   });
-  const { write: paidMint } = useContractWrite(paidMintConfig);
+  const { writeContract: paidMint } = useWriteContract();
 
-  const usedFreeMintResponse = useContractRead({
+  const usedFreeMintResponse = useReadContract({
     // TODO: the chainId should be dynamic
     address: contractAddress,
     abi: contract.abi,
     functionName: 'usedFreeMints',
     args: [address],
-    enabled: (address?.length ?? 0) > 0,
-    watch: true, // Watch for changes in the data and update the state if they use their free mint
```

Comment:
if you want to replicate `watch` feature with wagmi 2. This is how we did it

```
import { QueryKey, useQueryClient } from '@tanstack/react-query';
import { useBlockNumber } from 'wagmi';
import { useEffect } from 'react';

type Props = {
  queryKey: QueryKey | undefined;
  watchBlockFrequency?: number;
  enabled?: boolean;
};

/**
 * Refetch wagmi contract hook when current block number satisfies condition Number(blockNumber) % watchBlockFrequency === 0
 * @param queryKey - key returned from wagmi hook
 * @param watchBlockFrequency - Used to determine how often to refetch. Default is 1 (Number(blockNumber) % watchBlockFrequency === 0)
 * @param enabled
 */
export const useRefetchEveryNBlock = ({
  queryKey,
  watchBlockFrequency = 1,
  enabled = false,
}: Props) => {
  const queryClient = useQueryClient();
  const { data: blockNumber } = useBlockNumber({ watch: enabled, query: { enabled } });

  // Achieves 'watch' mode. Refetch on block change if Number(blockNumber) % watchBlockFrequency === 0
  useEffect(() => {
    if (enabled && blockNumber && Number(blockNumber) % watchBlockFrequency === 0)
      queryClient.invalidateQueries({ queryKey });
  }, [blockNumber, queryClient, enabled, queryKey, watchBlockFrequency]);
};
```

And then for example add use like so 

```
const {
    data: erc20Data,
    isLoading,
    isSuccess: isSuccessReadContracts,
    isFetching,
    isError,
    refetch,
    queryKey,
  } = useBalance({
    address: isEnabled ? userAccount : undefined,
    chainId,
    query: { enabled: isEnabled },
  });

  useRefetchEveryNBlock({ queryKey, enabled: isEnabled && watch, watchBlockFrequency });
```

Reply:
This is good to know! I think watch is not needed for us right no but good to know we can do this in wagmi 2.0 as well. Great solution!